service/history/decisionChecker.go

```diff
@@ -64,7 +64,7 @@ type (
 )
 
 const (
-	reservedTaskListPrefix = "/__cadence_sys/"
```

Comment:
Not part of this PR but this const seems to be defined in multiple places. Needs to be extracted somewhere.

Reply:
Agreed.  Will do it in a separate PR.